### PR TITLE
Add fleetworkspace mutating webhook

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: pkg/resources/fleetworkspace/fleetworkspace.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+					Controller:         new(bool),
+					BlockOwnerDeletion: new(bool),
+				},
+			},
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:     "User",
+				APIGroup: "rbac.authorization.k8s.io",
+				Name:     request.UserInfo.UID,
```
Comment (ibuildthecloud) [inline pkg/resources/fleetworkspace/fleetworkspace.go:141]: Should be User, not UID

File: pkg/resources/fleetworkspace/fleetworkspace.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+					"fleetworkspaces",
+				},
+				ResourceNames: []string{
+					fw.Name,
+				},
+			},
+		},
+	}
+
+	rolebindingView := rbacv1.ClusterRoleBinding{
```
Comment (ibuildthecloud) [inline pkg/resources/fleetworkspace/fleetworkspace.go:91]: Create RoleBindings, not clusterrolebindings.  We want the role deleted when the namespace is.

File: pkg/resources/fleetworkspace/fleetworkspace.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+					APIVersion:         management.GroupName,
+					Name:               fw.Name,
+					UID:                fw.UID,
+					Controller:         new(bool),
+					BlockOwnerDeletion: new(bool),
+				},
+			},
+		},
+	}
+	if _, err := m.namespaces.Create(&namespace); err != nil && !errors.IsAlreadyExists(err) {
```
Comment (ibuildthecloud) [inline pkg/resources/fleetworkspace/fleetworkspace.go:55]: We should fail on "IsAlreadyExists"

**Comment by StrongMonkey** [comment]
Close in favor of https://github.com/rancher/webhook/pull/14

**Pull request closed without merging**